### ISSUE TITLE
This is a non-recoverable error: When creating Item with nested query

### ISSUE THREAD

**Issue by Sumis34** [prisma/prisma#15064]
Hi Prisma Team! My Prisma Client just crashed. This is the report:

When I want to create a `Deal` my Prisma throws this error. After this, I have to restart my DB to get any Query response.

## Versions

| Name            | Version            |
|-----------------|--------------------|
| Node            | v16.15.1           | 
| OS              | windows            |
| Prisma Client   | 4.2.1              |
| Query Engine    | 2920a97877e12e055c1333079b8d19cee7f33826|
| Database        | mysql              |

## Error
```
parent result: Some(ManyRecords { records: [Record { values: [String("cl7d7nqfz022480v5t3mul30x"), String("Foto Kurs"), String("cl7c7s68p0008isv5fcw5b2cf"), DateTime(2022-08-28T10:50:46.574+00:00), DateTime(2022-08-28T10:50:46.574+00:00), String("Take cool shots"), Int(2), DateTime(2022-08-28T10:50:47.327+00:00), Boolean(false), Enum("course"), Int(2), Float(BigDecimal("-4.01966")), Float(BigDecimal("22.50279"))], parent_id: None }], field_names: ["id", "title", "authorId", "start", "end", "description", "allowedMemberCount", "createdAt", "isRequest", "type", "categoryId", "lat", "long"] }), relation: Relation { name: "author", model_a_name: "Rating", model_b_name: "User", model_a: OnceCell((Weak)), model_b: OnceCell((Weak)), field_a: OnceCell((Weak)), field_b: OnceCell((Weak)), manifestation: Inline(InlineRelation { in_table_of_model_name: "Rating" }), internal_data_model: "#InternalDataModelWeakRef#" }
```

## Logs
```
yEngine:loader Searching for Query Engine Library in H:\Code\choeimir\node_modules\.prisma\client
prisma:client:libraryEngine:loader loadEngine using H:\Code\choeimir\node_modules\.prisma\client\query_engine-windows.dll.node
prisma:client:libraryEngine sending request, this.libraryStarted: true
prisma:client:libraryEngine sending request, this.libraryStarted: false
prisma:client:libraryEngine library starting
prisma:client:libraryEngine library started
prisma:client:request_handler {"code":"P2003","clientVersion":"4.2.1","meta":{"field_name":"categoryId"}}
prisma:tryLoadEnv Environment variables loaded from H:\Code\choeimir\.env
prisma:client dirname H:\Code\choeimir\node_modules\.prisma\client
prisma:client relativePath ..\..\..\prisma
prisma:client cwd H:\Code\choeimir\prisma
prisma:client clientVersion 4.2.1
prisma:client clientEngineType library
prisma:client:libraryEngine internalSetup
prisma:client:libraryEngine:loader Searching for Query Engine Library in H:\Code\choeimir\node_modules\.prisma\client
prisma:client:libraryEngine:loader loadEngine using H:\Code\choeimir\node_modules\.prisma\client\query_engine-windows.dll.node
prisma:client:libraryEngine sending request, this.libraryStarted: true
prisma:client:libraryEngine sending request, this.libraryStarted: false
prisma:client:libraryEngine library starting
prisma:client:libraryEngine library started
prisma:client:request_handler {"code":"P2003","clientVersion":"4.2.1","meta":{"field_name":"categoryId"}}
prisma:tryLoadEnv Environment variables loaded from H:\Code\choeimir\.env
prisma:client dirname H:\Code\choeimir\node_modules\.prisma\client
prisma:client relativePath ..\..\..\prisma
prisma:client cwd H:\Code\choeimir\prisma
prisma:client clientVersion 4.2.1
prisma:client clientEngineType library
prisma:client:libraryEngine internalSetup
prisma:client:libraryEngine:loader Searching for Query Engine Library in H:\Code\choeimir\node_modules\.prisma\client
prisma:client:libraryEngine:loader loadEngine using H:\Code\choeimir\node_modules\.prisma\client\query_engine-windows.dll.node
prisma:client:libraryEngine sending request, this.libraryStarted: true
prisma:client:libraryEngine sending request, this.libraryStarted: false
prisma:client:libraryEngine library starting
prisma:client:libraryEngine library started
prisma:tryLoadEnv Environment variables loaded from H:\Code\choeimir\.env
prisma:client dirname H:\Code\choeimir\node_modules\.prisma\client
prisma:client relativePath ..\..\..\prisma
prisma:client cwd H:\Code\choeimir\prisma
prisma:client clientVersion 4.2.1
prisma:client clientEngineType library
prisma:client:libraryEngine internalSetup
prisma:client:libraryEngine:loader Searching for Query Engine Library in H:\Code\choeimir\node_modules\.prisma\client
prisma:client:libraryEngine:loader loadEngine using H:\Code\choeimir\node_modules\.prisma\client\query_engine-windows.dll.node
prisma:client:libraryEngine sending request, this.libraryStarted: true
prisma:client:libraryEngine sending request, this.libraryStarted: false
prisma:client:libraryEngine library starting
prisma:client:libraryEngine library started
prisma:tryLoadEnv Environment variables loaded from H:\Code\choeimir\.env
prisma:client dirname H:\Code\choeimir\node_modules\.prisma\client
prisma:client relativePath ..\..\..\prisma
prisma:client cwd H:\Code\choeimir\prisma
prisma:client clientVersion 4.2.1
prisma:client clientEngineType library
prisma:client:libraryEngine internalSetup
prisma:client:libraryEngine:loader Searching for Query Engine Library in H:\Code\choeimir\node_modules\.prisma\client
prisma:client:libraryEngine:loader loadEngine using H:\Code\choeimir\node_modules\.prisma\client\query_engine-windows.dll.node
prisma:client:libraryEngine sending request, this.libraryStarted: true
prisma:client:libraryEngine sending request, this.libraryStarted: false
prisma:client:libraryEngine library starting
prisma:client:libraryEngine library started
prisma:tryLoadEnv Environment variables loaded from H:\Code\choeimir\.env
prisma:client dirname H:\Code\choeimir\node_modules\.prisma\client
prisma:client relativePath ..\..\..\prisma
prisma:client cwd H:\Code\choeimir\prisma
prisma:client clientVersion 4.2.1
prisma:client clientEngineType library
prisma:client:libraryEngine internalSetup
prisma:client:libraryEngine:loader Searching for Query Engine Library in H:\Code\choeimir\node_modules\.prisma\client
prisma:client:libraryEngine:loader loadEngine using H:\Code\choeimir\node_modules\.prisma\client\query_engine-windows.dll.node
prisma:client:libraryEngine sending request, this.libraryStarted: true
prisma:client:libraryEngine sending request, this.libraryStarted: false
prisma:client:libraryEngine library starting
prisma:client:libraryEngine library started
```

## Client Snippet
Weirdly in this create statement authorId is required but not when creating another entry (rating), that also has an author property. 
```ts
  const deal = await prisma.deal.create({
    data: {
      allowedMemberCount: 2,
      type: "course",
      description: "Take cool shots",
      end: new Date(),
      start: new Date(),
      lat: -4.01966,
      long: 22.50279,
      title: "Foto Kurs",
      category: {
        connect: {
          id: 3,
        },
      },
      authorId: session.user.id, //session.user.id,
      author: {
        connect: {
          id: session.user.id,
        },
      },
    },
    include: {
      author: true,
      category: true,
      price: true,
    },
  });
```

## Schema
```prisma
generator client {
    provider = "prisma-client-js"
}

datasource db {
    provider = "mysql"
    url      = env("DATABASE_URL")
}

model Account {
    id                String  @id @default(cuid())
    userId            String
    type              String
    provider          String
    providerAccountId String
    refresh_token     String? @db.Text
    access_token      String? @db.Text
    expires_at        Int?
    token_type        String?
    scope             String?
    id_token          String? @db.Text
    session_state     String?
    user              User    @relation(fields: [userId], references: [id], onDelete: Cascade)

    @@unique([provider, providerAccountId])
}

model Session {
    id           String   @id @default(cuid())
    sessionToken String   @unique
    userId       String
    expires      DateTime
    user         User     @relation(fields: [userId], references: [id], onDelete: Cascade)
}

model User {
    id             String    @id @default(cuid())
    name           String?
    email          String?   @unique
    emailVerified  DateTime?
    image          String?
    isVerified     Boolean   @default(false)
    role           Role      @default(user)
    memberOf       Deal[]
    reports        Report[]
    recevedRatings Rating[]  @relation(name: "receivedRatings")
    ratings        Rating[]  @relation(name: "author")
    deals          Deal[]    @relation(name: "author")
    Account        Account[]
    Session        Session[]
}

model VerificationToken {
    identifier String
    token      String   @unique
    expires    DateTime

    @@unique([identifier, token])
}

model Deal {
    id                 String    @id @default(cuid())
    title              String
    authorId           String
    author             User      @relation(fields: [authorId], references: [id], name: "author", onDelete: Cascade)
    start              DateTime?
    end                DateTime?
    description        String    @db.Text
    allowedMemberCount Int?
    createdAt          DateTime  @default(now())
    isRequest          Boolean   @default(false)
    type               DealType
    categoryId         Int
    lat                Float
    long               Float
    category           Category  @relation(fields: [categoryId], references: [id])
    members            User[]
    price              Price[]
    files              File[]
}

model Category {
    id    Int    @id @default(autoincrement())
    name  String @unique
    deals Deal[]
}

model Report {
    id        Int      @id @default(autoincrement())
    authorId  String
    author    User     @relation(fields: [authorId], references: [id], onDelete: Cascade)
    createdAt DateTime @default(now())
    message   String
}

model Rating {
    id        Int      @id @default(autoincrement())
    authorId  String
    author    User     @relation(fields: [authorId], references: [id], name: "author", onDelete: Cascade)
    userId    String
    user      User     @relation(fields: [userId], references: [id], name: "receivedRatings", onDelete: Cascade)
    rating    Int
    createdAt DateTime @default(now())
}

model Price {
    id        Int    @id @default(autoincrement())
    price     Float
    extention String
    currency  String
    dealId    String
    deal      Deal   @relation(fields: [dealId], references: [id], onDelete: Cascade)
}

model File {
    id     String     @id @default(cuid())
    key    String
    dealId String
    deal   Deal       @relation(fields: [dealId], references: [id])
    access FileAccess @default(public)
}

enum FileAccess {
    private
    public
}

enum Role {
    admin
    user
}

enum DealType {
    service
    course
}

```

## Prisma Engine Query
```
{"X":{}}
```

**Comment by Jolg42** [comment]
Can reproduce in `4.2.1` and `4.3.1` when `author: true,` is part of the `include`, removing `author: true,` result in a successful query.

with
```ts
import { PrismaClient } from "@prisma/client";

async function main() {
  const prisma = new PrismaClient();

  const userId = "1234";
  const user = await prisma.user.create({
    data: {
      id: userId,
    },
  });
  console.log({ user });

  const category = await prisma.category.create({
    data: {
      id: 3,
      name: "test",
    },
  });
  console.log({ category });

  const deal = await prisma.deal.create({
    data: {
      allowedMemberCount: 2,
      type: "course",
      description: "Take cool shots",
      end: new Date(),
      start: new Date(),
      lat: -4.01966,
      long: 22.50279,
      title: "Foto Kurs",
      category: {
        connect: {
          id: 3,
        },
      },
      // This is invalid (Type Error) in our internal dev version (so will be invalid in the next release)
      authorId: userId,
      author: {
        connect: {
          id: userId,
        },
      },
    },
    include: {
      author: true,
      category: true,
      price: true,
    },
  });
  console.log({ deal });

  prisma.$disconnect();
}

main();

```

outputs
```
ts-node main.ts
thread 'tokio-runtime-worker' panicked at 'parent result: Some(ManyRecords { records: [Record { values: [String("cl7rgmhnj0000yl4p7sv1asgn"), String("Foto Kurs"), String("1234"), DateTime(2022-09-07T10:10:32.233+00:00), DateTime(2022-09-07T10:10:32.233+00:00), String("Take cool shots"), Int(2), DateTime(2022-09-07T10:10:32.287+00:00), Boolean(false), Enum("course"), Int(3), Float(BigDecimal("-4.01966")), Float(BigDecimal("22.50279"))], parent_id: None }], field_names: ["id", "title", "authorId", "start", "end", "description", "allowedMemberCount", "createdAt", "isRequest", "type", "categoryId", "lat", "long"] }), relation: Relation { name: "author", model_a_name: "Rating", model_b_name: "User", model_a: OnceCell((Weak)), model_b: OnceCell((Weak)), field_a: OnceCell((Weak)), field_b: OnceCell((Weak)), manifestation: Inline(InlineRelation { in_table_of_model_name: "Rating" }), internal_data_model: "#InternalDataModelWeakRef#" }', query-engine/core/src/interpreter/query_interpreters/nested_read.rs:254:9
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
/Users/j42/Repros/15064/node_modules/@prisma/client/runtime/index.js:29917
      throw new PrismaClientRustPanicError(message, this.client._clientVersion);
            ^
PrismaClientRustPanicError: 
Invalid `prisma.deal.create()` invocation in
/Users/j42/Repros/15064/main.ts:22:34

  19 // });
  20 // console.log({ category });
  21 
→ 22 const deal = await prisma.deal.create(
parent result: Some(ManyRecords { records: [Record { values: [String("cl7rgmhnj0000yl4p7sv1asgn"), String("Foto Kurs"), String("1234"), DateTime(2022-09-07T10:10:32.233+00:00), DateTime(2022-09-07T10:10:32.233+00:00), String("Take cool shots"), Int(2), DateTime(2022-09-07T10:10:32.287+00:00), Boolean(false), Enum("course"), Int(3), Float(BigDecimal("-4.01966")), Float(BigDecimal("22.50279"))], parent_id: None }], field_names: ["id", "title", "authorId", "start", "end", "description", "allowedMemberCount", "createdAt", "isRequest", "type", "categoryId", "lat", "long"] }), relation: Relation { name: "author", model_a_name: "Rating", model_b_name: "User", model_a: OnceCell((Weak)), model_b: OnceCell((Weak)), field_a: OnceCell((Weak)), field_b: OnceCell((Weak)), manifestation: Inline(InlineRelation { in_table_of_model_name: "Rating" }), internal_data_model: "#InternalDataModelWeakRef#" }
```

Using our internal dev version `4.4.0-dev.30` the issue is gone, the query is successful.

Note: the query needs to be modified, as passing `authorId` results in a Type Error 
```ts
  const deal = await prisma.deal.create({
    data: {
      allowedMemberCount: 2,
      type: "course",
      description: "Take cool shots",
      end: new Date(),
      start: new Date(),
      lat: -4.01966,
      long: 22.50279,
      title: "Foto Kurs",
      category: {
        connect: {
          id: 3,
        },
      },
      // This is invalid (Type Error) in our internal dev version (so will be invalid in the next release)
      // authorId: userId,
      author: {
        connect: {
          id: userId,
        },
      },
    },
    include: {
      author: true,
      category: true,
      price: true,
    },
  });
  console.log({ deal });
```

```
ts-node main.ts
{
  deal: {
    id: 'cl7rgqdvz0000dd4p82zr2mm6',
    title: 'Foto Kurs',
    authorId: '1234',
    start: 2022-09-07T10:13:33.970Z,
    end: 2022-09-07T10:13:33.970Z,
    description: 'Take cool shots',
    allowedMemberCount: 2,
    createdAt: 2022-09-07T10:13:34.030Z,
    isRequest: false,
    type: 'course',
    categoryId: 3,
    lat: -4.01966,
    long: 22.50279,
    author: {
      id: '1234',
      name: null,
      email: null,
      emailVerified: null,
      image: null,
      isVerified: false,
      role: 'user'
    },
    category: { id: 3, name: 'test' },
    price: []
  }
}
```

So it seems this is already fixed and will be available in the next version (4.4.0)

**Comment by Jolg42** [comment]
Duplicate of https://github.com/prisma/prisma/issues/14696
Fixed by https://github.com/prisma/prisma-engines/pull/3169
